### ISSUE TITLE
breaking-change: sort-imports via import specifier 

### ISSUE THREAD

**Issue by PHILLIPS71** [eslint/eslint#14735]
**What rule do you want to change?**
sort-imports

**Does this change cause the rule to produce more or fewer warnings?**
Potentially more with the addition of the mixed memberSyntaxSortOrder option

**How will the change be implemented? (New option, new default behavior, etc.)?**
The rule would accept a different array of ```memberSyntaxSortOrder``` options those being ["none", "all", "named", "mixed", "default"]. Each option will no longer be classified by the number of members it is importing, but by the import specifier type. 

An additional option of mixed would be added that would handle a mixture between default and named specifier types.

**Please provide some example code that this change will affect:**

```js
import bravo from "bravo";
import { alpha } from "alpha";
import { charlie, delta } from "charlie";
import delta, { gamma } from 'delta.js';
```

**What does the rule currently do for this code?**
It would group imports via their import specifier types, then order each group. Previously this would fail due to the fact that ```import { alpha } from "alpha";``` would need to be above ```import bravo from "bravo";``` as it would be a single and started with A.

This change would reclassify the following imports, where as none and all options would remain the same
```js
import bravo from "bravo"; // default
import { alpha } from "alpha"; //named
import { charlie, delta } from "charlie"; //named
import delta, { gamma } from 'delta.js'; // mixed
```

**What will the rule do after it's changed?**
The rule would group then sort each import by the import specifier type, not by the number of members it is importing. I personally don't like mixing of default and single named imports, I'd much rather have them grouped then sorted.

**Are you willing to submit a pull request to implement this change?**
Yes, I'll make some time to implement this

**Comment by mdjermanovic** [comment]
Hi @PHILLIPS71, thanks for the issue!

The team has made the difficult decision to freeze stylistic rules (please find more details in [this blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies)), so we won't be adding any new options or changing the existing ones in the `sort-imports` rule.

I'm closing this issue per our [rules policies](https://github.com/eslint/eslint#stylistic-rule-updates).

> The rule would group then sort each import by the import specifier type, not by the number of members it is importing. I personally don't like mixing of default and single named imports, I'd much rather have them grouped then sorted.

This was already requested several times, and it really makes a lot of sense. It would be best if someone could create a new rule with a different set of options and publish the rule in a plugin.
